src/concepts/layout/README.md

```diff
@@ -0,0 +1,216 @@
+    .constraints(vec![
+        Constraint::Percentage(50),
+        Constraint::Percentage(50),
+    ])
+    .split(f.size());
+
+let inner_layout = Layout::default()
+    .direction(Direction::Horizontal)
+    .constraints(vec![
+        Constraint::Percentage(25),
+        Constraint::Percentage(75),
+    ])
+    .split(outer_layout[1]);
+```
+
+In this situation, the terminal window is initially split vertically into two equal parts by
+`outer_layout`. Then, `inner_layout` splits the second rectangle of `outer_layout` horizontally,
+creating two areas that are 25% and 75% of the width of the original rectangle, respectively.
+
+Rendering some Paragraphs of text into the above layouts produces the following:
+
+```rust
+frame.render_widget(Paragraph("outer 0").borders(Borders::ALL), outer_layout[0]);
+frame.render_widget(Paragraph("inner 0").borders(Borders::ALL), inner_layout[0]);
+frame.render_widget(Paragraph("inner 1").borders(Borders::ALL), inner_layout[1]);
+```
+
+```raw
+┌───────────────────────────────────┐
+│outer 0                            │
+│                                   │
+│                                   │
+└───────────────────────────────────┘
+┌────────────────┐┌─────────────────┐
+│inner 0         ││inner 1          │
+│                ││                 │
+│                ││                 │
+└────────────────┘└─────────────────┘
+```
+
+This enables you to divide the terminal window into multiple sections of varying sizes, giving you
+the flexibility to create complex and adaptive graphical interfaces.
+
+## Constraints
+
+Constraints dictate the size and arrangement of components within layouts. The Ratatui framework
+provides several constraint types for fine-tuning your user interface's layout:
+
+- `Constraint::Length(u16)`: This constraint specifies a specific number of rows or columns that a
+  rectangle should take up. Note that this is determined by absolute size and is not responsive to
+  the overall terminal window size.
+
+- `Constraint::Percentage(u16)`: This constraint offers a size relative to the size of the parent
+  layout or the terminal window itself. For instance, `Constraint::Percentage(50)` signifies that a
+  rectangle should take up half of its parent's size.
+
+- `Constraint::Ratio(u16, u16)`: Utilizing ratios offers an even finer granularity for splitting
+  your layout. For instance, `Constraint::Ratio(1, 3)` will allocate 1/3rd of the parent's size to
+  this constraint.
+
+- `Constraint::Min(u16)`: Immerses a minimum limit to the size of a component. If a `Min` constraint
+  is ensured with a `Percentage` or `Ratio`, the component will never shrink below the specified
+  minimum size.
+
+- `Constraint::Max(u16)`: Limits the maximum size of a component. Similar to `Min`, if mixed with
+  `Percentage` or `Ratio`, the component will never exceed the specified maximum size.
+
+```admonish warning
+The Ratio and Percentage are defined in terms of the parent's size
```

Comment:
```suggestion
The `Ratio` and `Percentage` are defined in terms of the parent's size.
```

Reply:
I think probably not on this one as  those are concepts rather than code - perhaps lower case would be more sensible.

Reply:
Actually - changed

Reply:
Makes sense!